thundra/plugins/metric/metric_plugin.py

```diff
@@ -21,92 +22,116 @@ def __init__(self):
         self.system_cpu_total_end = float(0)
         self.process_cpu_usage_start = float(0)
         self.process_cpu_usage_end = float(0)
-        self.stat_data = {}
+        self.metric_data = {}
         self.reporter = {}
+        self.tracer = ThundraTracer()
 
     def before_invocation(self, data):
         self.reporter = data['reporter']
         context = data['context']
-        stat_time = time.time() * 1000
-        self.stat_data = {
-            'transactionId': data['transactionId'],
-            'applicationName': getattr(context, 'function_name', None),
+        metric_time = time.time() * 1000
+        function_name = getattr(context, constants.CONTEXT_FUNCTION_NAME, None)
+
+        active_span = self.tracer.get_active_span()
+
+        self.metric_data = {
+
+            'type': "Metric",
+            'agentVersion': '',
+            'dataModelVersion': constants.DATA_FORMAT_VERSION,
             'applicationId': utils.get_application_id(context),
+            'applicationDomainName':active_span.domain_name if active_span is not None else '',
+            'applicationClassName': active_span.class_name if active_span is not None else '',
+            'applicationName': function_name,
             'applicationVersion': getattr(context, constants.CONTEXT_FUNCTION_VERSION, None),
-            'applicationProfile': utils.get_environment_variable(constants.THUNDRA_APPLICATION_PROFILE, ''),
-            'applicationType': 'python',
-            'functionRegion': utils.get_environment_variable(constants.AWS_REGION, ''),
-            'statTimestamp': int(stat_time)
+            'applicationStage': '',
```

Comment:
should be taken from environment variable `thundra_agent_lambda_application_stage`